### PR TITLE
Add ORT minimal with NNAPI EP to Android CI

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Add ORT minimal with NNAPI EP to Android CI

**Motivation and Context**
- The added build/test to Android CI will only run UT, additional onnx_test_runner with customer .ort models will be added later

### CODE REVIEW

**Comment by gwang-msft** [comment]
A level 1 optimized mnist.onnx was added to make UT pass since it will need Gemm op to be included, and since we are using our own required_operators.config and cannot use the testdata\reduced_ops_via_config.config which contains "Gemm"

File: tools/ci_build/github/linux/ort_nnapi_minimal/nnapi_minimal_build_full_ort_and_create_ort_files.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# This script will run a full ORT build and use the python package built to generate ort format test files,
+# and the exclude ops config file, which will be used in the build_minimal_ort_and_run_tests.sh
+
```
Comment (skottmckay) [inline tools/ci_build/github/linux/ort_nnapi_minimal/nnapi_minimal_build_full_ort_and_create_ort_files.sh:5]: Are we duplicating too much stuff here vs. adding to the Android CI? Won't that just add more work to the CI servers, so it's better to have one longer CI than to add another one that duplicates some of the work?

Comment (gwang-msft) [reply tools/ci_build/github/linux/ort_nnapi_minimal/nnapi_minimal_build_full_ort_and_create_ort_files.sh]: Move the minimal build and run UT to Android CI
Will add some onnx_test_runner case later

**Review by skottmckay — DISMISSED** [review]

**Review by skottmckay — DISMISSED** [review]

**Comment by gwang-msft** [comment]
There are 3 build step now in the Andoid CI
1. Build full ORT with CPU EP only
2. Build full ORT with NNAPI EP
3. Build minimal ORT with NNAPI EP

No matter how I tried, the CI will only pass with any 2 combination of the above 3, the only way to make it pass is to not build java in step 3

**Review by skottmckay — APPROVED** [review]

**Pull request merged**